physics/body_centred_non_rotating_dynamic_frame_body.hpp

```diff
@@ -4,22 +4,45 @@
 #include "physics/body_centred_non_rotating_dynamic_frame.hpp"
 
 #include "geometry/identity.hpp"
+#include "geometry/rotation.hpp"
 
 namespace principia {
 namespace physics {
 namespace internal_body_centred_non_rotating_dynamic_frame {
 
 using geometry::AngularVelocity;
 using geometry::Identity;
+using geometry::OrthogonalMap;
+using geometry::Rotation;
 
 template<typename InertialFrame, typename ThisFrame>
 BodyCentredNonRotatingDynamicFrame<InertialFrame, ThisFrame>::
-BodyCentredNonRotatingDynamicFrame(
-    not_null<Ephemeris<InertialFrame> const*> const ephemeris,
-    not_null<MassiveBody const*> const centre)
+    BodyCentredNonRotatingDynamicFrame(
+        not_null<Ephemeris<InertialFrame> const*> const ephemeris,
+        not_null<MassiveBody const*> const centre)
     : ephemeris_(ephemeris),
       centre_(centre),
-      centre_trajectory_(ephemeris_->trajectory(centre_)) {}
+      centre_trajectory_(ephemeris_->trajectory(centre_)),
+      orthogonal_map_(
+          [this]() {
+            // Note that we cannot do this by making |equatorial| and
+            // |biequatorial| virtual members of |MassiveBody|, because that
+            // class is not templatized on |InertialFrame|.
+            auto const rotating_body =
+                dynamic_cast<RotatingBody<InertialFrame> const*>(&*centre_);
+            if (rotating_body == nullptr) {
+              return Identity<InertialFrame, ThisFrame>().Forget();
+            }
+            // In coordinates, the third parameter is |polar_axis|, but we seem
+            // to be a bit confused as to which of these things should be
+            // vectors or bivectors here.
+            // TODO(egg): Figure that out.
```

Comment:
It feels reasonable to use a bivector for the axis.  I think we have a similar "confused" comment somewhere else.